src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -360,7 +360,7 @@ private boolean inventoryNeedsDump()
         return (worker.getCitizenInventoryHandler().isInventoryFull()
                   || job.getActionsDone() >= getActionsDoneUntilDumping()
                   || wantInventoryDumped())
-                 && !(job instanceof JobDeliveryman);
+                 && !(job instanceof JobDeliveryman) && getState() != INVENTORY_FULL;
```

Comment:
this getState() != INVENTORY_FULL, wouldn't it be better at the beginning of the function (to avoid having to calculate all the other things when the inv is full already)

Reply:
Yep

Reply:
indeed